src/commands/auth.gleam

```diff
@@ -1,12 +1,117 @@
 //// Command logic for the authentication tool.
 
+import gleam/int
 import gleam/io
+import gleam/option.{None, Some}
 import glint.{type CommandInput, type Glint}
+import plex_pin_auth
+import plex_pin_auth/util/parser.{type PlexError, type PlexPin}
+import repeatedly
+import util/client_identifier
 import util/commands/auth_tool
+import util/env.{ConfigOption}
+import util/terminal
+
+/// The interval in which we poll the server for the token
+const interval_ms = 2000
 
 /// The 'auth' command logic
 fn do(_input: CommandInput) {
-  io.println("todo")
+  // First validate the config. If it's not valid then this command can not be executed
+  // This command has a reliance on hostname and username
+  case env.is_base_config_valid() {
+    False -> {
+      io.println(
+        "You have no setup the config yet! Please run `plexrpc config` first.",
+      )
+      terminal.exit()
+    }
+    _ -> Nil
+  }
+
+  // Second validate the auth config.
+  // If there is already a token we need to confirm with the user that they want to replace the current token
+  case env.is_auth_config_valid() {
+    True ->
+      case
+        terminal.confirm(
+          "You already have an auth token. Do you want to replace it?",
+          default: Some(False),
+        )
+      {
+        True -> {
+          // Spacing out the terminal if the prompt was sent
+          io.println("")
+        }
+        False -> terminal.exit()
+      }
+
+    _ -> Nil
+  }
+
+  // Generate the client identifier
+  let assert Ok(client_id) = client_identifier.generate()
+
+  // We need to create the Plex pin and inform the user of the next steps
+  let create_pin_response =
+    plex_pin_auth.create_pin(client_id)
+    |> handle_plex_error
+
+  // Extract the ok response since we know it's okay to go now :)
+  let assert Ok(created_pin) = create_pin_response
```

Comment:
We don't need to assert this. If `handle_plex_error` kicks you out of the terminal when this fails then we can just have it return the ok response.